src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -215,8 +215,20 @@ private async Task InitializeMEFAsync()
                 await DeleteOnRestartManager.Value.DeleteMarkedPackageDirectoriesAsync(ProjectContext.Value);
             }
 
-            ProjectRetargetingHandler = new ProjectRetargetingHandler(_dte, SolutionManager.Value, this, componentModel);
-            ProjectUpgradeHandler = new ProjectUpgradeHandler(this, SolutionManager.Value);
+            IVsTrackProjectRetargeting vsTrackProjectRetargeting = await AsyncServiceProvider.GlobalProvider.GetServiceAsync<SVsTrackProjectRetargeting, IVsTrackProjectRetargeting>();
```

Comment:
AsyncServiceProvider is probably redundant here, because NuGetPackage extends AsyncPackage, which itself is the provider.

Reply:
Ok. Good to know. Addressed.